### DISCUSSION TITLE
I am experiencing challenges with the Soil Condition Model

### DISCUSSION THREAD

**Discussion by Achidago** [SERVIR-WA/GALUP#96] [Q&A] [answered]
I get this feedback when i create the soil model and try running it. 

"Traceback (most recent call last):
File "C:\Users\silac\AppData\Roaming\QGIS\QGIS3\profiles\default\processing\scripts\weighted_sum.py", line 128, in processAlgorithm
input_gdf[output_clm] = utils.weighted_sum(input_gdf, fields_weights)
File "C:\Users\silac\AppData\Roaming\Python\Python37\site-packages\pylusat\utils.py", line 223, in weighted_sum
return df[[*col_weights]].dot(wgt_df)
File "C:\PROGRA~1\QGIS3~1.10\apps\Python37\lib\site-packages\pandas\core\frame.py", line 1140, in dot
np.dot(lvals, rvals), index=left.index, columns=other.columns
File "<__array_function__ internals>", line 6, in dot
TypeError: unsupported operand type(s) for *: 'NoneType' and 'float'

Error encountered while running Weighted Sum of Fields
Error encountered while running Weighted Sum of Fields
Execution failed after 25.94 seconds

Loading resulting layers
The following layers were not correctly generated.<ul><li>C:\Users\silac\AppData\Local\Temp/processing_1710f20b9be748d19caf363a49d764c8/3d166b07ff37447fb3cb2774d55baee7/WeightedSum.gpkg</li></ul>You can check the 'Log Messages Panel' in QGIS main window to find more information about the execution of the algorithm."

[Soil Condition.zip](https://github.com/SERVIR-WA/GALUP/files/6780247/Soil.Condition.zip)
@chjch

**Comment by chjch** [comment]
Hi Silas, 

Can you please attach your model by dragging it below your question (just like how you did it in the exercises)?

**Comment by chjch** [comment, marked as answer]
Hi Silas,

The problem is with how you use **Weighted Sum of Fields**. Please note that in the **graphical modeler** you should use a **semicolon**, i.e., **_;_**, as the _**fields delimiter**_. No official documentation was found for this particular behavior, but it is shared many times in the [GIS Stack Exchange](https://gis.stackexchange.com) community, such as [this post](https://gis.stackexchange.com/a/294703/83211).

![Image 12](https://user-images.githubusercontent.com/12037254/124837348-a7c5c300-df52-11eb-8d85-44004010d79c.png)

Also, you probably want to set the default value for the weights you used, which was missing in your model.

![Image 13](https://user-images.githubusercontent.com/12037254/124839059-10626f00-df56-11eb-8576-aa5c5d59256e.png)

**Reply by Achidago** [reply]
It worked. Thanks

**Reply by chjch** [reply]
Thank you Silas. Please mark the above as **answer**.
